src/vfs2.c

```diff
@@ -403,6 +404,37 @@ struct shm {
 static_assert(sizeof(struct shm_region) == VFS2_WAL_INDEX_REGION_SIZE,
 	      "shm regions have the expected size");
 
+/**
+ * Allocate a new shm region at the next-highest index.
+ *
+ * Returns a pointer to the new region, or NULL if allocation failed.
+ * In the latter case the shm is unchanged.
+ */
+static struct shm_region *shm_grow(struct shm *shm)
+{
+	int index = shm->num_regions;
+	struct shm_region *r = sqlite3_malloc64(VFS2_WAL_INDEX_REGION_SIZE);
+	if (r == NULL) {
+		goto err;
+	}
+	*r = (struct shm_region){};
+	sqlite3_uint64 size =
+	    (sqlite3_uint64)(index + 1) * (sqlite3_uint64)sizeof(*shm->regions);
```

Comment:
I am curious, does SQLite increase the region count 1 by 1 or does it do something else like double the amount? Obviously not in this PR and not that important actually.

Reply:
Good question, as far as I know SQLite only maps regions on-demand in response to needing more space for the WAL-index, so it grows by one region at a time. We assert this in both the old and new VFS in xShmUnmap.